src/Profiles.ts

```diff
@@ -234,103 +281,53 @@ export class Profiles {
 
     public loadNamedProfile(name: string, type?: string): IProfileLoaded {
         for (const profile of this.allProfiles) {
-            if (profile.name === name && (type ? profile.type === type : true)) {
-                return profile;
-            }
+            if (profile.name === name && (type ? profile.type === type : true)) { return profile; }
         }
-        throw new Error(localize("loadNamedProfile.error.profileName", "Could not find profile named: ")
-            + name + localize("loadNamedProfile.error.period", "."));
+        throw new Error(localize("loadNamedProfile.error.profileName", "Could not find profile named: {0}.", name));
     }
 
-    public getDefaultProfile(type: string = "zosmf"): IProfileLoaded {
-        return this.defaultProfileByType.get(type);
-    }
-
-    public getProfiles(type: string = "zosmf"): IProfileLoaded[] {
-        return this.profilesByType.get(type);
-    }
+    public getProfiles(type: string = "zosmf"): IProfileLoaded[] { return this.profilesByType.get(type); }
 
     public async refresh(): Promise<void> {
         this.allProfiles = [];
         this.allTypes = [];
+
+        // Set the default base profile (base is not a type included in registeredApiTypes)
+        let profileManager = await this.getCliProfileManager("base");
+        if (profileManager) {
+            try {
+                DefaultProfileManager.getInstance().setDefaultProfile("base", (await profileManager.load({ loadDefault: true })));
+            } catch (err) {
+                if (err.message !== `No default profile set for type "base"`) {
```

Comment:
Currently, I am still seeing the error about `No default profile set for type "base"`.
I think it would be more robust to use something like a return code rather than the error string to check this, as strings are more prone to changing, but apparently the returned error code is `undefined`, so I understand why it is done this way.

However, the error string in the `if` condition needs a period at the end of it for the string to match exactly and the condition to return `true`:
<img width="345" alt="Screen Shot 2020-09-14 at 8 43 30 AM" src="https://user-images.githubusercontent.com/45975633/93087534-af844600-f666-11ea-9c81-7d02468274fa.png">